File: playground/csharp/FibonacciNumbers.cs
using System;

namespace Program
{
    class FibonacciNumbers
    {
        static int Main()
        {
            int sequence_length;
            int x = 1;  // x and y are the first two numbers of the Fibonacci sequence.
            int y = 1;
            int z;

            Console.Write("How many Fibonacci numbers should be seen? > ");
            sequence_length = Convert.ToInt32(Console.ReadLine());

            Console.WriteLine($"\nPrinting {sequence_length} Fibonacci numbers...\n");
            Console.WriteLine($"[1] {x}\n[2] {y}");

            for (int i = 2; i < sequence_length; i++)
            {
                z = x + y;
                x = y;
                y = z;
                Console.WriteLine($"[{i + 1}] {z}");
            }

            return 0;
        }
    }
}

File: playground/lua/fibonacci_numbers.lua
-- Get the next fibonacci number from x and y.
local function nextFibonacciNumber(x, y)
    return x + y
end

local function main()
    io.write("How many fibonacci numbers do you want to see? > ")
    local nums = io.read('n')

    -- Set up the first three fibonacci numbers.
    local x = 1
    local y = 1
    local z = nextFibonacciNumber(x, y)
    local n = 0

    io.write("\nListing " .. nums .. " Fibonacci numbers...\n\n")

    if nums >= 1 then
        n = n + 1
        io.write('[' .. n .. '] ' .. x .. '\n')
    end

    if nums >= 2 then
        n = n + 1
        io.write('[' .. n .. '] ' .. y .. '\n')
    end

    if nums >= 3 then
        n = n + 1
        io.write('[' .. n .. '] ' .. z .. '\n')
    end

    while n < nums do
        x = y
        y = z
        z = nextFibonacciNumber(x, y)
        n = n + 1
        io.write('[' .. n .. '] ' .. z .. '\n')
    end
    return 0
end

os.exit(main())
